src/test/java/com/google/sps/servlets/ContestServletTest.java

```diff
@@ -0,0 +1,122 @@
+package com.google.sps.servlets;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalURLFetchServiceTestConfig;
+import java.io.PrintWriter;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.*;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class ContestServletTest {
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig(), new LocalURLFetchServiceTestConfig());
+
+  @Mock HttpServletRequest httpServletRequest;
+  @Mock HttpServletResponse httpServletResponse;
+  @Mock PrintWriter printWriter;
+
+  private static Set<Long> candidateIdSet;
+  private static Entity electionEntityOne;
+  private static Entity contestEntityOne;
+  private static Entity contestEntityTwo;
+
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @BeforeClass
+  public static void classSetUp() {
+    helper.setUp();
+
+    candidateIdSet = new HashSet<Long>();
+    candidateIdSet.add(Long.valueOf(1));
+    candidateIdSet.add(Long.valueOf(2));
+    candidateIdSet.add(Long.valueOf(3));
+
+    electionEntityOne = new Entity("Election");
```

Comment:
I would initialize these in a static block, and then just copy them using `new Entity().setPropertiesFrom(...)` when you want to use them in the tests.

https://www.geeksforgeeks.org/g-fact-79/

Reply:
You can also just initialize it directly in the unit test. Having too much setup code hurts readability, which is extra important in tests. Err on the side of DAMP not DRY:

https://drive.google.com/file/d/15GGGANZoaG9JIyGyrTkZWkmYlaih01MJ/view?usp=sharing
(let me know if you don't have permission for this link)

Reply:
Placed entity set up code in setUp method tagged with @Before